pkg/authn/principal_test.go

```diff
@@ -0,0 +1,56 @@
+package authn
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPrincipal(t *testing.T) {
+	// Arrange
+
+	// using current time for a bit of fuzzing
```

Comment:
So while this works, it took a bit of mental gymnastics to consider how/why it works. I'd prefer something a bit more readable, like the anonymous principal struct on lines 30-35, than the generated regular eua user.

Reply:
Made a couple changes that I hope make it more legible, while also keeping some of the dynamic nature (whereas the `anonymous` principal is a static).

Reply:
yep, more readable!